libs/server/Objects/ItemBroker/CollectionItemBroker.cs

```diff
@@ -0,0 +1,578 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Tsavorite.core;
+
+namespace Garnet.server
+{
+    /// <summary>
+    /// This class brokers collection items for blocking operations.
+    /// When a supported blocking command is initiated, RespServerSession will call the GetCollectionItemAsync method
+    /// with the desired object type and operation and a list of keys to the desired objects.
+    /// When an item is added to a collection, the StorageSession will call the Publish method with the relevant object key
+    /// to notify the broker that a new item may be available.
+    /// The main loop, in the Start method, listens for published item additions as well as new observers
+    /// and notifies the calling method if an item was found.
+    /// </summary>
+    public class CollectionItemBroker : IDisposable
+    {
+        // Queue of events to be handled by the main loops
+        private readonly AsyncQueue<BrokerEventBase> brokerEventsQueue = new();
```

Comment:
can these heavy components be instantiated lazily? i.e. if user is not using blocking collections, we should not incur either allocations or tasks.

Reply:
same holds for all the other allocations below.